src/scripts/sdk/ProtocolAPI/index.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  IFractalProtocolAPI,
+  IFact,
+  IAsyncIteratorCreator,
+} from "@pluginTypes/index";
+
+import SingletonConnection from "@sdk/InpageProvider/connection";
+import ConnectionTypes from "@models/Connection/types";
+
+class FactIterator implements AsyncIterator<any> {
+  private i: number = 0;
+  private factCount: Promise<number>;
+  private factsAPI: FactsAPI;
+  constructor(factsAPI: FactsAPI) {
```

Comment:
Newline between instance variables and constructor please. Also between each function definition.